File: frontend/src/lib/auth/domain.ts
export const COMPANY_EMAIL_DOMAIN = "simplicreative.com";

/** Return true only for *@simplicreative.com (case-insensitive). */
export function isCompanyEmail(email: string | null | undefined): boolean {
  if (!email || typeof email !== "string") return false;
  const trimmed = email.trim();
  const at = trimmed.lastIndexOf("@");
  if (at <= 0 || at === trimmed.length - 1) return false;
  const domain = trimmed.slice(at + 1).toLowerCase();
  return domain === COMPANY_EMAIL_DOMAIN;
}

File: backend/app/core/domain.py
from __future__ import annotations

COMPANY_EMAIL_DOMAIN = "simplicreative.com"


def is_company_email(email: str | None) -> bool:
    """Return True only for *@simplicreative.com (case-insensitive)."""
    if not email or not isinstance(email, str):
        return False
    trimmed = email.strip()
    if "@" not in trimmed:
        return False
    local, _, domain = trimmed.rpartition("@")
    if not local or not domain:
        return False
    return domain.lower() == COMPANY_EMAIL_DOMAIN
